libs/ledger-live-common/src/device/use-cases/appDataBackup/deleteAppDataUseCase.ts

```diff
@@ -0,0 +1,22 @@
+import { from, Observable } from "rxjs";
+import { AppName, AppStorageType, DeleteAppDataEvent, StorageProvider } from "./types";
+import { DeviceModelId } from "@ledgerhq/devices";
+
+/**
+ * Delete the local app data for a specific app on a Ledger device.
+ *
+ * @param appName name of the app to delete data for
+ * @param deviceModelId model id of the device
+ * @param storageProvider storage provider object used for storing the backup data
+ * @param deleteAppDataFn function that returns observable for the delete process
+ * @returns Observable<DeleteAppDataEvent>
+ */
+export function deleteAppDataUseCase(
+  appName: AppName,
+  deviceModelId: DeviceModelId,
+  storageProvider: StorageProvider<AppStorageType>,
+  deleteAppDataFn: () => Observable<DeleteAppDataEvent>,
+) {
+  const obs: Observable<DeleteAppDataEvent> = from(deleteAppDataFn());
+  return obs;
```

Comment:
[COULD] Remove unused parameters, and do this directly:
```
return deleteAppDataFn()
```

Reply:
OR it should be like that:
```
export function deleteAppDataUseCase(
  appName: AppName,
  deviceModelId: DeviceModelId,
  storageProvider: StorageProvider<AppStorageType>,
  deleteAppDataFn: (appName, deviceModelId, storageProvider) => Observable<DeleteAppDataEvent>,
) {
  return deleteAppDataFn(appName, deviceModelId, storageProvider
}
```

Reply:
I don't know actually just keep it like that maybe 🤷‍♂️

Reply:
I think that @ofreyssinet-ledger 's proposition is correct, the objectif of this layer is to add the storage specification. The result of the execution of `deleteAppDataFn` is already an observable of type `DeleteAppDataEvent` which should be exposed directly to the components (user interface logic).